modules/platform/nuxeo-binary-metadata/src/main/java/org/nuxeo/binary/metadata/internals/MetadataRuleDescriptor.java

```diff
@@ -108,4 +115,11 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    @Override
+    public int compareTo(MetadataRuleDescriptor o) {
+        return Comparator.comparing(MetadataRuleDescriptor::getOrder)
+                         .thenComparing(MetadataRuleDescriptor::getId)
+                         .compare(this, o);
```

Comment:
Actually we could even put `Comparator.comparing(MetadataRuleDescriptor::getOrder).thenComparing(MetadataRuleDescriptor::getId)` in a constant but not worth it here given the rare usage

Reply:
Actually rules are sorted on the fly when checking a filter, so maybe it's worth optimizing this?

Reply:
Ah yes maybe then